### PR TITLE
Cleanup Various ActionListener and -Runnable Usages

### PR DESCRIPTION
**Pull request by original-brownbear**
* Dry up code for creating simple `ActionRunnable` a little
* Shorten some other code around `ActionListener` usage, in particular
when wrapping it in a `TransportResponseListener`

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/bwc
Jenkins run elasticsearch-ci/packaging-sample

**Review by ywelsch — APPROVED** [review]
LGTM.
@andrershov can you have a look here too?

**Comment by original-brownbear** [comment]
@andrershov ping

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/2

**watcher**

**Comment by original-brownbear** [comment]
Jenkins run elasticsearch-ci/bwc

File: server/src/main/java/org/elasticsearch/action/ActionRunnable.java [hunk 1]
```diff
@@ -29,6 +30,23 @@
...
 
+    /**
+     * Creates a {@link Runnable} that wraps the given listener and a consumer of it that is executed when the {@link Runnable} is run.
+     * Invokes {@link ActionListener#onFailure(Exception)} on it if an exception is thrown on executing the consumer.
+     * @param listener ActionListener to wrap
+     * @param consumer Consumer of wrapped {@code ActionListener}
+     * @param <T> Type of the given {@code ActionListener}
+     * @return Wrapped {@code Runnable}
+     */
+    public static <T> ActionRunnable<T> wrap(ActionListener<T> listener, CheckedConsumer<ActionListener<T>, Exception> consumer) {
```
Comment (andrershov) [inline server/src/main/java/org/elasticsearch/action/ActionRunnable.java]: I don't immediately see how this method makes our code easier to understand.
I realize that now we can use lambdas, but I don't think that the resulting code is easier to grasp.
So I vote for reverting this change back.

**Review by andrershov — COMMENTED** [review]
Personally, I don't like adding `wrap` method to ActionRunnable. But I don't strongly hold the opinion.

File: server/src/main/java/org/elasticsearch/action/ActionRunnable.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/main/java/org/elasticsearch/action/ActionRunnable.java]: You can shorten all these spots https://github.com/elastic/elasticsearch/pull/42273/files#diff-a42a639093633662ea658bec29044a4cR1043 and save a bunch of lines when imo. We have a bunch of similar short-cuts for listeners as well so I find this a pretty natural simplification.

**Comment by original-brownbear** [comment]
@andrershov good to merge?

**Review by andrershov — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks @andrershov + @ywelsch !

**Pull request merged**